lib/Onyx.js

```diff
@@ -979,17 +1003,29 @@ function merge(key, changes) {
         .then((existingValue) => {
             try {
                 // We first only merge the changes, so we can provide these to the native implementation (SQLite uses only delta changes in "JSON_PATCH" to merge)
-                const batchedChanges = applyMerge(mergeQueue[key]);
+                const batchedChanges = applyMerge(undefined, mergeQueue[key]);
 
                 // Clean up the write queue so we
                 // don't apply these changes again
                 delete mergeQueue[key];
 
                 // After that we merge the batched changes with the existing value
-                const modifiedData = applyMerge([batchedChanges], existingValue);
+                let modifiedData = applyMerge(existingValue, [batchedChanges]);
+
+                // For objects, we want to remove all top level keys that are null
```

Comment:
Again, please modify the comment to explain _why_ we want to remove all top level keys that are null.

> For objects, the key for null values needs to be removed from the object to ensure the value will get removed from storage completely.